### PR TITLE
176950724 apigw auth

### PR DESCRIPTION
**Pull request by corydodt**
- Add an authorizer module that can be deployed to AWS Lambda and used as a custom authorizer for API gateway

TBD before finishing:

- [x] update bats tests to create and submit a real token instead of the non-working one
- [x] finish unit test coverage
- [x] additional update to the infrastructure repo to grant secretsmanager access to the authorizer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ the agent, and its dependencies, as well.
 1. Start by building the lambda zipfile:
 
     ```#!bash
-    make -B function.zip
+    make -B function.zip function-jawthorizer.zip
```
Comment (jamesbeedy) [inline README.md:80]: Haha

File: README.md [hunk 2]
```diff
@@ -88,15 +88,16 @@ install `terraform`.
...
     prod, edge, or other).
 
-    You will need to set one environment variable before running terraform:
+    Before running terraform:
 
     ```#!bash
-    # a path to step 1 function.zip
-    export TF_VAR_zipfile=/some/path/function.zip
-    ```
+    # create scratch.auto.tfvars
```
Comment (jamesbeedy) [inline README.md:94]: where does the `scratch.auto.tfvars` file live?

**Review by jamesbeedy — APPROVED** [review]
`jawthorizer` module + readme + tests + helpers, oh my!

Excellent work on this well rounded pr including all the goodies.

I've read through it and deployed it multiple times, its pretty air tight. Nice work.

👍

File: README.md (hunk 2 above)
Comment (corydodt) [reply README.md:94]: That section says to be in `live/license-manager/xxxx` right above that.

**Pull request merged**